AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests+Support.swift

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import Amplify
+import AWSMobileClient
+import AWSAPICategoryPlugin
+@testable import AWSAPICategoryPluginTestCommon
+@testable import AmplifyTestCommon
+import AWSPluginsCore
+
+extension GraphQLAuthDirectiveIntegrationTests {
+    func createNote(_ id: String = UUID().uuidString,
+                    content: String) -> Result<MutationSyncResult, GraphQLResponseError<MutationSyncResult>> {
+        let createdNoteInvoked = expectation(description: "note was created")
+        var result: Result<MutationSyncResult, GraphQLResponseError<MutationSyncResult>>?
+        let note = SocialNote(id: id, content: content, owner: nil)
+        let request = GraphQLRequest<MutationSyncResult>.createMutation(of: note)
+        _ = Amplify.API.mutate(request: request, listener: { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let note):
+                    result = .success(note)
+                case .failure(let error):
+                    result = .failure(error)
+                }
+                createdNoteInvoked.fulfill()
+            case .failed(let error):
+                XCTFail("Got failed, error: \(error)")
+            default:
+                XCTFail("Unexpected event: \(event)")
+            }
+        })
+
+        wait(for: [createdNoteInvoked], timeout: TestCommonConstants.networkTimeout)
+        return result!
```

Comment:
Instead of using force unwrap how about something like..
```
guard let result = result else {
    XCTFail("Failed on creating note")
}
return result
```
Seems like it would be a friendlier error in the event that we couldn't get a result? And of course, if we do this here, we can apply this to the rest of the functions in this file.

Reply:
i wanted the return value to be easy to manage for the tests to use, which makes it hard to return something in the guard that is not coming from asyncEvent's completed case. I can use guard here and throw a fatalError because it's unexpected, which is equivalent of the force cast